sdk/storage/Azure.Storage.Blobs/README.md

```diff
@@ -33,77 +44,131 @@ Blob storage is designed for:
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+// Get a connection string to our Azure Storage account.  You can
+// obtain your connection string from the Azure Portal (click
+// Access Keys under Settings in the Portal Storage account blade)
+// or using the Azure CLI with:
+//
+//     az storage account show-connection-string --name <account_name> --resource-group <resource_group>
+//
+// And you can provide the connection string to your application
+// using an environment variable.
+string connectionString = "<connection_string>";
+
+// Get a reference to a container named "sample-container" and then create it
+BlobContainerClient container = new BlobContainerClient(connectionString, "sample-container");
+container.Create();
+
+// Get a reference to a blob named "sample-file" in a container named "sample-container"
+BlobClient blob = container.GetBlobClient("sample-file");
+
+// Open a file and upload it's data
+using (FileStream file = File.OpenRead("local-file.jpg"))
 {
-    await blob.UploadAsync(data);
+    blob.Upload(file);
 }
 ```
 
 ### Downloading a blob
+
 ```c#
-string connectionString = <connection_string>;
-var service = new BlobServiceClient(connectionString);
-var container = service.GetBlobContainerClient("mycontainer");
-var blob = container.GetBlockBlobClient("myblob");
-
-var download = await blob.DownloadAsync();
-using (var file = File.Create("BlockDestination.txt"))
-{   
-    await download.Value.Content.CopyToAsync(file);
+// Get a reference to the public blob at https://aka.ms/bloburl
+BlobClient blob = new BlobClient(new Uri("https://aka.ms/bloburl"));
+
+// Download the blob
+BlobDownloadInfo download = blob.Download();
+using (FileStream file = File.OpenWrite("hello.jpg"))
+{
+    download.Content.CopyTo(file);
 }
 ```
 
 ### Enumerating blobs
+
 ```c#
-string connectionString = <connection_string>;
-var service = new BlobServiceClient(connectionString);
-var container = service.GetBlobContainerClient("mycontainer");
+string connectionString = "<connection_string>";
+
+// Get a reference to a container named "sample-container"
+BlobContainerClient container = new BlobContainerClient(connectionString, "sample-container");
 
-string marker;
-do
+// List all of its blobs
+foreach (BlobItem blob in container.GetBlobs())
 {
-    var response = await container.ListBlobsFlatSegmentAsync(marker);
-    foreach (var blob in response.Value.BlobItems)
-    {
-        Console.WriteLine(blob.Name);
-    }
-    marker = response.Value.NextMarker;
+    Console.WriteLine(blob.Name);
 }
-while (!string.IsNullOrEmpty(marker));
 ```
 
+### Authenticating with Azure.Identity
+
+The [Azure Identity library](identity) provides easy Azure Active Directory support for authentication.
+
+```c#
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+
+// Create a BlobServiceClient that will authenticate through Active Directory
+Uri accountUri = new Uri("https://MYSTORAGEACCOUNT.blob.core.windows.net/");

```

Comment:
Might be worth adding a comment saying that they will need the Uri of the storage account they previously created/ want to consume...